src/Http/Client/Response.php

```diff
@@ -243,6 +243,16 @@ public function isOk()
         return in_array($this->code, $codes);
     }
 
+    /**
+     * Check if the response was in the 2xx/3xx range.
+     *
+     * @return bool
+     */
+    public function isSuccessful()
+    {
+        return $this->code >= 200 && $this->code <= 308;
```

Comment:
Hm I'd go `return $this->code >= 200 && $this->code <= 399;` to be future proof and work with non standard http codes that still respect the basic ranged (1xx, 2xx, 3xx, 4xx and 5xx).

Also the naming is bad. Clearly only 2xx is "success".
From the RFC: 

```

   2.  Any 2xx (Successful) response to a CONNECT request implies that
       the connection will become a tunnel immediately after the empty
       line that concludes the header fields.  A client MUST ignore any
       Content-Length or Transfer-Encoding header fields received in
       such a message.
```
https://tools.ietf.org/html/rfc7230

So is successful should be:
`return $this->code >= 200 && $this->code <= 299;`

Reply:
This method can stay, but should be renamed to something else.

Reply:
@ionas Please see the discussion in the original ticket.
The method name is fine IMO.

Reply:
As long as https://github.com/cakephp/cakephp/blob/master/src/TestSuite/IntegrationTestCase.php#L727 is not adjusted either we can probably keep the current code here as it is.
Having them in sync makes responses and their checks consistent.

Reply:
@dereuromark it is not fine, instead it is completely wrong. HTTP clearly defines what is successful and what is not. Why make things work in unexpected ways?

Suggestion: Instead `isOK` could be updated to be 200-399?

Reply:
As discusses before BC outweighs any such changes or HTTP synchr.
So instead we keep things as all already discussed - and can re-evaluate that in a major jump.
Again: see also assertResponseSuccess()/assertResponseOk()

Reply:
Every new, well versed developer, will expect `isSuccessful()` to be 2xx range and without reading the source code bugs (and security issues) will happen.

One thing is not being able to change `isOk()`, the other is making yet another step to be non-standardish and use `isSuccessful`- instead maybe one of these, or we/you can come up with a better name:
* `->is2xx3xx()`
* `->is200to399()`
* `->isSuccessfulOrRedirect()`
* `->isGood()` (I know cheesy)
* `->isOkay()` (and deprecate `isOk()` right away)

Reply:
isSuccessfulOrRedirect() could work for the 3.x series.

Reply:
Aren't redirects a type of 'success'? Having the same term used in same way in the framework is one advantage of leaving the method as is.

Reply:
Sure they are similar. That's the reason isOk should/could return true on 200-399 (in 4.x)?

Reply:
Then we don't need this method.

Reply:
I would prefer not adding any new method and just modify `isOk()` in 4.x based on consensus.